pkg/steps/script_step.go

```diff
@@ -158,24 +168,29 @@ tar zxvf %s.tgz 1>&2
 	}
 
 	if c.Image != "" {
-		dockerArgs := []string{}
-		for _, v := range c.Volumes {
-			dockerArgs = append(dockerArgs, "-v", v)
-		}
-		if c.Envfile != "" {
-			dockerArgs = append(dockerArgs, "--env-file", c.Envfile)
-		} else if context.Autoenv() {
+		if context.Autoenv() {
 			autoEnv, err := context.GenerateAutoenv()
 			if err != nil {
-				log.Errorf("script step failed to generate autoenv: %v", err)
+				log.Errorf("script step failed to generate autoenv with docker run: %v", err)
 			}
-			for name, value := range autoEnv {
-				if value == "" {
-					continue
-				}
-				dockerArgs = append(dockerArgs, "--env", fmt.Sprintf("%s=%s", name, value))
+			if c.Env == nil {
+				c.Env = make(map[string]string, len(autoEnv))
+			}
+			for k, v := range autoEnv {
+				c.Env[k] = v
 			}
 		}
+
+		dockerArgs := []string{}
+		for _, v := range c.Volumes {
+			dockerArgs = append(dockerArgs, "-v", os.ExpandEnv(v))
```

Comment:
Thanks for preserving `ExpandEnv` here!

As far as I remember, this was necessary to expand the `HOME` env var contained in the src of volume mount, that is useful for mounting `.aws` or `.ssh` on the host machine to the container.